plugins/www/Optimove.js

```diff
@@ -1,44 +1,204 @@
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "registerUser", [
+        userId,
+        userEmail,
+      ]);
+    });
+  },
 
-    getVisitorId: function () {
-        return new Promise((resolve, reject) => { exec(resolve, reject, 'OptimoveSDKPlugin', 'getVisitorId', []); });
+  getVisitorId: function () {
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "getVisitorId", []);
+    });
+  },
 
-    },
+  getCurrentUserIdentifier: function () {
+    return new Promise((resolve, reject) => {
+      exec(
+        resolve,
+        reject,
+        "OptimoveSDKPlugin",
+        "getCurrentUserIdentifier",
+        []
+      );
+    });
+  },
 
-    getCurrentUserIdentifier: function () {
-        return new Promise((resolve, reject) => { exec(resolve, reject, 'OptimoveSDKPlugin', 'getCurrentUserIdentifier', []); });
+  pushRegister: function () {
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "pushRegister", []);
+    });
+  },
 
-    }
-}
-module.exports = Optimove;
\ No newline at end of file
+  inAppUpdateConsent: function (consented) {
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "inAppUpdateConsent", [
+        consented,
+      ]);
+    });
+  },
+
+  inAppGetInboxItems: function () {
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "inAppGetInboxItems", []);
+    });
+  },
+  inAppMarkAllInboxItemsAsRead: function () {
+    return new Promise((resolve, reject) => {
+      exec(
+        resolve,
+        reject,
+        "OptimoveSDKPlugin",
+        "inAppMarkAllInboxItemsAsRead",
+        []
+      );
+    });
+  },
+  inAppMarkAsRead: function (inAppInboxItem) {
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "inAppMarkAsRead", [
+        inAppInboxItem,
+      ]);
+    });
+  },
+  inAppGetInboxSummary: function () {
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "inAppGetInboxSummary", []);
+    });
+  },
+  inAppPresentInboxMessage: function (inAppInboxItem) {
+    return new Promise((resolve, reject) => {
+      exec(resolve, reject, "OptimoveSDKPlugin", "inAppPresentInboxMessage", [
+        inAppInboxItem,
+      ]);
+    });
+  },
+  inAppDeleteMessageFromInbox: function (inAppInboxItem) {
+    return new Promise((resolve, reject) => {
+      exec(
+        resolve,
+        reject,
+        "OptimoveSDKPlugin",
+        "inAppDeleteMessageFromInbox",
+        [inAppInboxItem]
+      );
+    });
+  },
+
+  setOnInboxUpdatedHandler: function (handler) {
+      inAppInboxUpdatedHandler = handler;
+  },
+  
+  setPushOpenedHandler(pushOpenedHandler) {
+    currentConfig["pushOpenedHandler"] = pushOpenedHandler;
+    checkIfPendingPushExists().then(success,error); 
```

Comment:
Not sure if this needs to chain a success/error callback?

Reply:
hmm although it's an async function?

Reply:
I removed the success and error